core/modules/views/plugins/views_plugin_style_grid.inc

```diff
@@ -31,41 +35,111 @@ class views_plugin_style_grid extends views_plugin_style {
    */
   function options_form(&$form, &$form_state) {
     parent::options_form($form, $form_state);
+    
+    // Flatten options to deal with the various hierarchy changes.
+    $options = views_grid_get_options($this->options);
```

Comment:
I suggest we make this a static method in this class rather than put it in views module. e.g.

```
static function flatten_options($options) {

}
```

Then call it with:

```
$options = views_plugin_style_grid::flatten_options($options);
```

Though honestly, perhaps it's better to just forgo flattening entirely and use the setting locations in the nested array.